File: e2e/playwright-typescript/tests/task-info/task-info.spec.ts
import { test } from '@/fixtures';
import { TaskContext } from '@/context/TaskContext';
import * as allure from "allure-js-commons";
import { AllureEpic } from '@/data/AllureEpic';

test.beforeAll(async () => {
  await allure.epic(AllureEpic.TASK_MANAGEMENT);
  await allure.feature('View task info');
});

test.describe('View task info', () => {
  let context: TaskContext;

  test.beforeEach(async ({ support }) => {
    context = new TaskContext();
    const response = context.createTaskResponse();

    await support.mock.api.getTasks([response]);
    await support.mock.api.getTask(response.id, response);
    await support.mock.api.getIsValid(response.id, true);
  });

  test('should view task info', async ({ step, validate }) => {
    // given
    await step.navigation.openMainPage();

    // when
    await step.tasks.openTaskInfoForm(context.title);

    // then
    await validate.task.data(context.createTaskData());
    await validate.task.isValid();
  });
});

File: e2e/capybara-ruby/spec/task_info/task_info_spec.rb
# frozen_string_literal: true

require 'spec_helper'

require_relative '../../context/task_context'
require_relative '../../data/allure_epic'

RSpec.describe 'View task info', type: :system,
                                 epic: AllureEpic::TASK_MANAGEMENT,
                                 feature: 'View task info',
                                 tms: '102' do
  let(:context) { TaskContext.new }

  before do
    response = context.create_task_response
    support.mock.api
         .get_tasks([response])
         .get_task(response[:id], response)
         .get_is_valid(response[:id], true)
  end

  it 'should view task info' do
    # given
    step.navigation.open_main_page

    # when
    step.tasks.open_task_info_form(context.title)

    # then
    validate.task.data(context.create_task_data)
    validate.task.is_valid
  end
end
